src/hoprd.rs

```diff
@@ -348,6 +349,24 @@ impl Hoprd {
     
 }
 
+    pub async fn notify_cluster(&self, context: Arc<ContextData>) -> Result<(), Error> {
+        let hoprd_namespace = self.metadata.namespace.as_ref().unwrap().to_owned();
+        let api: Api<ClusterHoprd> = Api::namespaced(context.client.clone(), &hoprd_namespace.to_owned());
+        if let Some(owner_reference) = self.owner_references().to_owned().first() {
+            if let Some(cluster) = api.get_opt(&owner_reference.name).await? {
+                if cluster.to_owned().status.unwrap().status != ClusterHoprdStatusEnum::Deleting {
+                    cluster.create_event(context.clone(), ClusterHoprdStatusEnum::OutOfSync, Some(self.name_any().to_owned())).await.unwrap();
+                    cluster.update_status(context.clone(), ClusterHoprdStatusEnum::OutOfSync).await.unwrap();
+                    println!("[INFO] Notifying ClusterHoprd {} that Hoprd node {} is being deleted", &owner_reference.name, self.name_any().to_owned())
+                }
+            } else {
+                println!("[WARN] ClusterHoprd {} not found", &owner_reference.name);
```

Comment:
Should this not be an error?

Reply:
I've made this improvement to create bidirectional communication between ClusterHoprd and Hoprd resource. The problem is that sometime we (or argoCD) might want to manually delete some Hoprd resources, and that will lead to a ClusterHoprd misconfiguration because it does not have all the required nodes running.
Regarding the WARN message it means that the Hoprd resource exists but his owner is not the ClusterHoprd which should not happen never. I added that code to be aware of that anomaly